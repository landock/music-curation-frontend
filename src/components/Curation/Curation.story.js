import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import Curation from '.';

const curation = {
  id: '23',
  imageUrl: 'https://placeimg.com/250/250/tech/grayscale',
  tracks: [
    {
      id: '12',
      imageUrl: 'https://placeimg.com/250/250/animals',
      artistName: 'spongebop',
      trackName: 'square wave',
    },
    {
      id: '13',
      imageUrl: 'https://placeimg.com/250/250/animals',
      artistName: 'Trevor Noah',
      trackName: 'Disko Deck',
    },
  ],
  name: 'story testing',
};
storiesOf('Curation', module).add('default', () => (
  <Curation curation={curation} />
));
