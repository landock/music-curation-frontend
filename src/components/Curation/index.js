import React, { Component } from 'react';
import { Item } from 'semantic-ui-react';

class Curation extends Component {
  render() {
    const { curation } = this.props;
    return !curation ? (
      <p>No Curfations</p>
    ) : (
      <Item>
        <Item.Image size="small" src={curation.imageUrl} />
        <Item.Content>
          <Item.Header>{curation.name}</Item.Header>
          <Item.Description>
            <p>{curation.description}</p>
            <strong>Tracks:</strong>{' '}
            <em>
              {curation.tracks &&
                curation.tracks
                  .map(track => {
                    return track.trackName;
                  })
                  .join(', ')}
            </em>
          </Item.Description>
        </Item.Content>
      </Item>
    );
  }
}

export default Curation;
