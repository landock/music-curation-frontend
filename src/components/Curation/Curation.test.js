import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import Curation from '.';

it('renders correctly', () => {
  let mockCuration = {
    id: 1,
    imageUrl: 'http://fpoimg.com/200x200',
    description: 'whatever description here',
    tracks: [
      {
        imageUrl: 'http://fpoimg.com/100x100',
        artistName: 'QA Dougie Test',
        trackName: 'testin shit',
        durationInSeconds: 123233,
        recordLabels: ['warner bros', 'heathcliff the cat and record label'],
      },
    ],
    name: 'Sprint 14 remix',
  };
  const component = shallow(<Curation curation={mockCuration} />);

  expect(component.text()).toEqual('<Item />');
});
