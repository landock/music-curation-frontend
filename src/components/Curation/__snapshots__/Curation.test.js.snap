// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Curation: default 1`] = `
<div
  className="item"
>
  <div
    className="ui small image"
    href={undefined}
  >
    <img
      src="http://fpoimg.com/200x200"
    />
  </div>
  <div
    className="content"
  >
    <div
      className="header"
    >
      Ergonomic Harber Pike
    </div>
    <div
      className="description"
    >
      <p>
        Try to input the GB array, maybe it will copy the haptic firewall!
      </p>
      <strong>
        Tracks:
      </strong>
       
      <em>
        clicks-and-mortar envisioneer portals, bleeding-edge iterate e-markets, 24/365 recontextualize models, front-end engineer vortals, granular synthesize applications, strategic brand e-tailers, virtual incentivize systems, value-added mesh architectures, extensible maximize experiences, mission-critical unleash metrics, robust e-enable communities, compelling enhance web services, user-centric monetize deliverables, e-business seize eyeballs, e-business exploit solutions, impactful redefine e-business, distributed expedite systems, best-of-breed maximize communities
      </em>
    </div>
  </div>
</div>
`;
